addons/payment_razorpay_oauth/models/payment_provider.py

```diff
@@ -0,0 +1,246 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+import pprint
+import requests
+import uuid
+
+from datetime import timedelta
+from hashlib import sha1
+from urllib.parse import urlencode, urljoin
+
+from odoo import _, fields, models
+from odoo.exceptions import RedirectWarning, ValidationError
+
+from odoo.addons.payment_razorpay import const
+from odoo.addons.payment_razorpay_oauth.const import OAUTH_URL
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentProvider(models.Model):
+    _inherit = 'payment.provider'
+
+    razorpay_key_id = fields.Char(required_if_provider=False)
+    razorpay_key_secret = fields.Char(required_if_provider=False)
+    razorpay_webhook_secret = fields.Char(required_if_provider=False)
+
+    # Oauth fields
+    razorpay_access_token = fields.Char(
+        string="Razorpay Access Token",
+        groups='base.group_system',
+        copy=False,
```

Comment:
In the OAuth flow, we block copying to avoid duplicating provider credentials (like the Razorpay account ID and public token) when the provider is duplicated to use two Razorpay account in Same DB.

Reply:
Ok, I see. However, I think that the opposite use-case of duplicating a provider to use *the same account* but with a different configuration is more common, so we should probably copy the fields and let the user reset the account if they want to create another account for some reason.